vitals/BatteryHealthFactor.java

```diff
@@ -0,0 +1,39 @@
+package vitals;
+
+public enum BatteryHealthFactor implements BatteryHealth {
+
+    TEMPERATURE {
+        @Override
+        public boolean isOutOfRange(float temperature, float soc, float chargeRate) {
+            boolean batteryHealthState = true;
+            if (temperature < 0 || temperature > 45) {
+                System.out.println("Temperature is out of range!");
+                batteryHealthState = false;
+            }
+            return batteryHealthState;
+        }
+    },
+    CHARGE_STATE {
+        @Override
+        public boolean isOutOfRange(float temperature, float soc, float chargeRate) {
+            boolean batteryHealthState = true;
+            if (soc < 20 || soc > 80) {
```

Comment:
The checks for temperature and SoC look the same. That would mean changes in the comparison logic will need to be done in two places.
Bring the comparison out into a function and reuse.